devtools/mockslack/usergroupslist.go

```diff
@@ -0,0 +1,47 @@
+package mockslack
+
+import (
+	"context"
+	"net/http"
+	"sort"
+)
+
+func (st *API) UserGroupList(ctx context.Context) ([]UserGroup, error) {
```

Comment:
Add comments to these matching the ones we have in `conversations.list`, for example